test-all-rest.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+set -e
+
+# Generates all REST APIs present in a google-api-dotnet-client repo
+# using Google.Api.Generator.Rest, and compares the results with the
+# Python-generated code.
+
+if [[ -z "$1" ]]
+then
+  echo "Required argument: path to root directory of google-api-dotnet-client repo"
+  exit 1
+fi
+
+declare -r CLIENT_REPO=$1
+
+rm -rf tmp
+mkdir tmp
+
+dotnet build -nologo -clp:NoSummary -v quiet Google.Api.Generator.Rest
+
+# Generate all the code with the C# generator
+for json in $CLIENT_REPO/DiscoveryJson/*.json
+do
+  echo "Generating $(basename $json)"
+  dotnet run --no-build -p Google.Api.Generator.Rest -- $json tmp/csharp
+done
+
+# Copy the Python-generated code
+for package_dir in tmp/csharp/*
+do
+  package=$(basename $package_dir)
+  if [[ ! -d $CLIENT_REPO/Src/Generated/${package} ]]
+  then
+    echo "Python directory for ${package} does not exist; deleting C# code"
```

Comment:
But also in the end, we'd need to check that we have generated everything that python had generated, right? Mostly to protect against the generator not propagating an exception somewhere I think.

Reply:
Yup - this only reduces unnecessary diffs due to things we choose not to generate in Python. Note that by now we've run the C# generator already, so if that throws, we'll have found it out already.